### PR TITLE
Docs now finally run with ToffeeIR master.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: docs/onnx_docs/Super-Resolution-tutorial.ipynb [hunk 1]
```diff
@@ -135,7 +133,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Exporting a model in PyTorch works via tracing. To export a model, you call the `torch.onnx.export()` function which executes the model, recording a trace of what operators are used to compute the outputs. Because `export` runs the model, we need provide an input tensor `x`. The values in this tensor are not important. It could be an image or a random tensor as long as it is the right size."
+    "Exporting a model in PyTorch works via tracing. To export a model, you call the `torch.onnx.export()` function. This will execute the model, recording a trace of what operators are used to compute the outputs. Because `export` runs the model, we need provide an input tensor `x`. The values in this tensor are not important; it can be an image or a random tensor as long as it is the right size."
```
Comment (prigoyal) [inline docs/onnx_docs/Super-Resolution-tutorial.ipynb:136]: we need provide an input tensor    -->   we need to provide an input tensor

File: docs/onnx_docs/Super-Resolution-tutorial.ipynb [hunk 2]
```diff
@@ -150,10 +148,10 @@
     "x = Variable(torch.randn(batch_size, 1, 224, 224), requires_grad=True)\n",
     "\n",
     "# Export the model \n",
-    "torch_out = torch.onnx.export(torch_model, # model being run \n",
-    "                              x, # the input to the model (or a tuple for multiple inputs) \n",
-    "                              \"super_resolution.onnx\", # where to save the model (can also be a file or file-like object)\n",
-    "                              export_params=True) # store the trained parameter weights inside the model file"
+    "torch.onnx.export(torch_model, # model being run \n",
```
Comment (prigoyal) [inline docs/onnx_docs/Super-Resolution-tutorial.ipynb:151]: this looks very ugly in git view, can we make it one liner and write all comments on top of this statement?

File: docs/onnx_docs/Super-Resolution-tutorial.ipynb [hunk 3]
```diff
@@ -190,12 +188,8 @@
...
     "# Construct a map from input names to Tensor data.\n",
     "# The graph itself contains inputs for all weight parameters, followed by the input image.\n",
-    "# Since the weights are already embedded, we just need to pass the input image, which is the \n",
-    "# last input the graph\n",
-    "W = {graph.input[-1]: x.data.numpy()}\n",
-    "\n",
-    "# Run the Caffe2 net:\n",
-    "c2_out = prepared.run(W)[0]\n",
+    "# Since the weights are already embedded, we just need to pass the input image.\n",
+    "c2_out = prepared.run(x.data.numpy())[0]\n",
```
Comment (prigoyal) [inline docs/onnx_docs/Super-Resolution-tutorial.ipynb:192]: I don't really like the "prepared" as variable name, can be renamed to "prepared_backend" or something shorter?

File: docs/source/onnx.rst [hunk 4]
```diff
@@ -141,6 +143,7 @@ list.  The operator set above is sufficient to export the following models:
 * SqueezeNet
 * SuperResolution
 * VGG
+* `word_language_model <https://github.com/pytorch/examples/tree/master/word_language_model>`_
```
Comment (prigoyal) [inline docs/source/onnx.rst:146]: I think the above list of ops is not complete, for example PRelu that I added for super-resolution from twitter. we should revisit and correct this

Comment (prigoyal) [reply docs/source/onnx.rst:146]: one another comment: I would perhaps also reorder the models like Vision models first, DCGAN, Super-Resolution, etc. just like we have separate repos for them in pytorch? not a big deal at all but just wanted to comment out.

**Review by prigoyal — CHANGES_REQUESTED** [review]
this doesn't run, the inputs to run() is incorrect, fixing now

File: docs/onnx_docs/Super-Resolution-tutorial.ipynb (hunk 3 above)
Comment (ezyang) [reply docs/onnx_docs/Super-Resolution-tutorial.ipynb:192]: Are you referring here? There's a ToffeeIR update in this PR which makes this syntax work.

File: docs/source/onnx.rst (hunk 4 above)
Comment (ezyang) [reply docs/source/onnx.rst:146]: I fixed this and pushed. I forgot to do the model reorder so I'll do that now.

**Comment by ezyang** [comment]
I'm going to obsolete the superres tutorial patches, since I don't know where they're supposed to go now.

**Pull request closed without merging**